design-principles.md

```diff
@@ -3,6 +3,7 @@
 ## Reusability
 - Existing features should be reused when possible instead of adding new ones. Before proposing a feature, try solving the problem with existing features first.
 - Prefer providing a solution in the Catalog when possible to adding a feature.
+- At authoring time (i.e. when authoring Pipelines and Tasks), authors should be able to include anything that is required for every execution of the Task or Pipeline. At run time (i.e. when invoking a Pipeline or Task via  PipelineRun or TaskRun), users should be able to control execution as needed by their context without having to modify Tasks and Pipelines. 
```

Comment:
kind of a nit but i think it would be helpful to be able to link to this directly (e.g. if im going to reference "runtime" in a doc) - i wonder if it would look terrible to make a subheading under reusability like `### Authoring time and Runtime`

(feel free to ignore! i can always link to the reusability section)

Reply:
the formatting was weird when i tried that, so I added numbering to the sections so you can reference the section and the relevant number